### PR TITLE
Yatopia 2 Electric Boogaloo [Yatopia Reforged]

### PR DESCRIPTION
**Pull request by duplexsystem**
This PR cleans up the mess that is our legacy code base by doing the following:
- Use Gradle and Toothpick for base build system based almost entirely off https://github.com/jmanpenilla/Toothpick
- Redo Yatopia's build system to be more dynamic, cleaner, and based in kotlin
- Include all of Purpur's patches

### CODE REVIEW

**Review by Titaniumtown — DISMISSED** [review]
LGTM

**Review by toinouH — DISMISSED** [review]
This system seems less complicated than the other one.

**Comment by HookWoods** [comment]
toothpick is really faster compared to the actual system, and you only have to compile the upstream dir on the first upstream or when you upstream. So really fast to apply patches, even for CI Server. Also, the new implementation for the sidestream looks pretty nice and works well. Maybe need one day to just found a fix for windows computer about the bash cmd that is not always recognized by IntelliJ but is by git bash. wp

**Review by HookWoods — DISMISSED** [review]
Yes

**Review by Titaniumtown — DISMISSED** [review]
yes

File: README.md
Comment (toinouH) [inline README.md]: Remove infos about recommended configs; we can re-ade it later.
```suggestion
 You can find a full explanation of the Yatopia configuration file on the [wiki](https://github.com/YatopiaMC/Yatopia/wiki). Check out the list of patches included in this project and who created them [here](PATCHES.md).
```

**Review by toinouH — COMMENTED** [review]
I understand if you want to leave it; we can do ir later too but it's confusing.

**Review by Titaniumtown — DISMISSED** [review]
review #3 lol

Edit: crap I referenced an issue xD

**Review by PhantomG27249 — DISMISSED** [review]
its an improvement to the build system etc, approving

**Review by toinouH — DISMISSED** [review]
Ready

**Review by HookWoods — DISMISSED** [review]

**Review by MrIvanPlays — DISMISSED** [review]
flying squids patch is present.
remove that

also update all the submodules while you're at it

**Comment by Titaniumtown** [comment]
@MrIvanPlays why would the flying squids patch prevent the merging of this PR? :confused:

**Comment by MrIvanPlays** [comment]
> @MrIvanPlays why would the flying squids patch prevent the merging of this PR? 😕

because flying squids are not gonna get into yatopia.
never ever ever ever

**Comment by Titaniumtown** [comment]
why

**Comment by Titaniumtown** [comment]
Stop acting like a child. If you don't want flying squids, you can disable it in `purpur.yml`, other than that it doesn't affect you. That is in no way a valid reason to block the merging of this PR. If you have an actual reason on why this new build system shouldn't be merged, feel free to tell us.

**Comment by MrIvanPlays** [comment]
> Stop acting like a child. If you don't want flying squids, you can disable it in `purpur.yml`, other than that it doesn't affect you. That is in no way a valid reason to block the merging of this PR. If you have an actual reason on why this new build system shouldn't be merged, feel free to tell us.

i have but ive dealt with it . if you dont like what im thinking then go somewhere else. I dont want the flying squids patch in yatopia, mainly cuz of historical reasons.

**Comment by MrIvanPlays** [comment]
also @duplexsystem if im not wrong purpur had a patch where it reverted the revert of mc-4 fix, blacklist that too

**Comment by Titaniumtown** [comment]
> i have but ive dealt with it . if you dont like what im thinking then go somewhere else. I dont want the flying squids patch in yatopia, mainly cuz of historical reasons.

Keyword **YOU** don't want it in yatopia, yatopia is not your project. And those "historical reasons" are stupid af.

**Comment by Titaniumtown** [comment]
> if you dont like what im thinking then go somewhere else

Everyone but you approves of this PR, it's the opposite situation here, you're in the minority. It's ***you*** who doesn't like what everyone else is thinking, not me or anyone else here.

**Comment by MrIvanPlays** [comment]
you can stop arguing you won't win.

**Comment by Titaniumtown** [comment]
If you have an actual reason as to not merging this PR, fell free to tell us. But as of now you can stop with this childish behavior.

**Comment by Titaniumtown** [comment]
Why did you request a review from me. lmfao

**Review by Titaniumtown — DISMISSED** [review]
lmfao, Ivan dismissed my review

**Comment by Titaniumtown** [comment]
> not a review team.

what? lmao

**Comment by Titaniumtown** [comment]
How am I not a part of the review team?

**Comment by MrIvanPlays** [comment]
> How am I not a part of the review team?

you're not removed from the org yet but you're not gonna have the perms you currently have.

**Comment by Titaniumtown** [comment]
I am a part of the org. And the only reason why I don't have the `review team` rank on the discord server is because you had a temper tantrum after I got it.

**Review by MrIvanPlays — DISMISSED** [review]
blacklist flying squids and revert of revert of mc-4 fix
update the submodules

**Comment by toinouH** [comment]
Ok guys stop.
Ivan make a proper review, only that matters: duplex will do the job for squids
Jettison: I know you strongly disagree his point of view but let him work.

**Comment by MrIvanPlays** [comment]
> I am a part of the org. And the only reason why I don't have the `review team` rank on the discord server is because you had a temper tantrum after I got it.

image

**Comment by Titaniumtown** [comment]
m8, you blurred like half the convo lmfao.

**Comment by MrIvanPlays** [comment]
> m8, you blurred like half the convo lmfao.

ofc i did, you will otherwise start arguing even more

**Comment by duplexsystem** [comment]
Ready for Merge.

**Review by Titaniumtown — DISMISSED** [review]
LGTM

**Review by krolik-exe — DISMISSED** [review]

**Comment by BillyGalbreath** [comment]
Funny, cause this system was designed exactly _for_ [Purpur](https://github.com/pl3xgaming/Purpur/pull/122). Why you trying to pass this off as your own?

**Comment by duplexsystem** [comment]
Screenshot from 2021-01-19 20-46-57
nope by my build system I just mean my the part that deals with other fork's patches

**Comment by duplexsystem** [comment]
also your forget it's not 100% your build system to being with you could say your taking toothpick and passing it off as your own right now I know you didn't mean it like that :)

**Comment by duplexsystem** [comment]
hope that clears things up @BillyGalbreath

**Comment by BillyGalbreath** [comment]
> also your forget it's not 100% your build system to being with you could say your taking toothpick and passing it off as your own right now I know you didn't mean it like that :)

We never claimed it was ours. We gave proper attribution, and even talked to both mikro _and_ mini while in the process of building it. You are flat out claiming it's yours, and even going as far as to say we could use it from you.. This is DMCA territory.

**Comment by katekerllenevich** [comment]
> my build system would be useful for purpur

purpur already has this lol, and you took it

**Comment by duplexsystem** [comment]
> I just mean my the part that deals with other fork's patches

if you want more attribution please ask :)

**Comment by duplexsystem** [comment]
> This is DMCA territory.

As far as I'm aware it's not because it's under MIT and you never included a Purpur or Pl3x specific copyright notice in your licence but again if you wish for more attribution I can give you it.

**Comment by duplexsystem** [comment]
@BillyGalbreath just so you know if you would like more attribution before merge now is the time to ask because we're merging soon.

**Comment by PaulBGD** [comment]
> > This is DMCA territory.
> 
> As far as I'm aware it's not because it's under MIT and you never included a Purpur or Pl3x specific copyright notice in your licence but again if you wish for more attribution I can give you it.

MIT doesn't give away the copyright, the original authors still own it. MIT just opens up some of these permissions so you can use it, not claim the copyright as your own.

**Comment by duplexsystem** [comment]
@PaulBGD I was just saying Purpur's licence does not have `(C) 2021 Pl3xgaming` or something so it's still being distributed under and MIT and this is still under MIT (so It should be compliment with the `The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.` clause), but regardless It was never my intention to claim all of toothpick as my own, just the fraction that I wrote and I'm willing to prove credit where ever @BillyGalbreath thinks is necessary.

**Comment by BillyGalbreath** [comment]
> Purpur's licence does not have `(C) 2021 Pl3xgaming` or something so it's still being distributed under and MIT and this is still under MIT (so It should be compliment with the `The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.` clause)

I suggest you read [Title 17 of the Copyright Law](https://www.copyright.gov/title17/title17.pdf) sections 401(a), 401(d), and 504(c)(2).

**Comment by duplexsystem** [comment]
I'm not here to have a legal battle. Do you want credit anywhere or do you want it taken down or neither? I've stated before I'm willing to work if you if you really care.

**Review by Titaniumtown — DISMISSED** [review]
LGTM, lets get this merged!!

**Comment by MrIvanPlays** [comment]
@BillyGalbreath @PaulBGD you have a point here, it's really not our system, it's a heavily modified toothpick to fit our needs. 
we still license it mit so i dont think there should be any issues with the license ( not a lawyer or something, but its the same license so whats the problem? )

**Comment by gabboman** [comment]
Im going to make my own fork with no dramas!



Fuck I had a discussion with myself I just failed.

**Review by MrIvanPlays — DISMISSED** [review]
Upstream commit message is not how it should be.

Paper and Purpur should be separated by a `/`.

There's literally NO way to resolve conflicting sidestream patches without messing up the rebuilt patches. EVERY known way of editing commits doesn't work ( they work only for our patches and not sidestream patches ). I don't know how will you fix this but if you can't find a fix this system gets scrapped.

Not sure if it is related, but I get dependency resolve problems whilst using IntelliJ.

File: patches/server/0002-Brandings.patch
Comment (Titaniumtown) [inline patches/server/0002-Brandings.patch]: uh oh, that shouldn't happen.

**Review by Titaniumtown — DISMISSED** [review]
the version in `0002-Brandings.patch` should be fixed

File: patches/server/0002-Brandings.patch
Comment (Titaniumtown) [reply patches/server/0002-Brandings.patch]: I'll fix it myself rq.

**Review by Titaniumtown — DISMISSED** [review]
LGTM

**Review by Titaniumtown — DISMISSED** [review]

**Review by Titaniumtown — DISMISSED** [review]
Now it should be read for merging.

**Review by MrIvanPlays — DISMISSED** [review]

**Comment by MrIvanPlays** [comment]
waiting on codemc to retrigger the build so i can merge

**Review by MrIvanPlays — APPROVED** [review]

**Review by toinouH — APPROVED** [review]
@MrIvanPlays

**Review by metabrixkt — APPROVED** [review]

**Pull request merged**
